rustls/src/server/mod.rs

```diff
@@ -146,6 +146,311 @@ impl<'a> ClientHello<'a> {
     }
 }
 
+/// Building a ServerConfig in a linker-friendly way.
```

Comment:
Since this is a decent amount of code, maybe it deserves its own `server::builder` module?